pkg/endpoint/endpoint.go

```diff
@@ -1583,9 +1607,9 @@ func (e UpdateStateChangeError) Error() string { return e.msg }
 // if there was an issue triggering policy updates for the given endpoint,
 // or if endpoint regeneration was unable to be triggered.
 func (e *Endpoint) Update(owner Owner, cfg *models.EndpointConfigurationSpec) error {
+	e.Mutex.Lock()
 	e.getLogger().WithField("configuration-options", cfg).Debug("updating endpoint configuration options")
 
-	e.Mutex.Lock()
```

Comment:
This locking change is no longer needed as `e.getLogger()` takes care of it's own locking.